TESTING.md

```diff
@@ -169,3 +169,66 @@ We have a suite of external snaps tests that runs each night using the latest sn
 This is an experimental suite, with still some details to define. The idea is to build a snap recording a manifest of all the details of the build. Then build the snap again, but this time using the manifest instead of the source `snapcraft.yaml`, and compare that both snaps are equal.
 
 Currently, the suite is using the snaps of the integration suite to check the reproducibility. It is located in https://github.com/elopio/snapcraft-reproducible/
+
+## Testing arm
+
+It is possible to emulate an arm64 machine on an amd64 host, which is very useful for running manual exploratory tests for snapcraft. To set it up:
+
+1. Download the latest ubuntu arm64 uefi image from https://cloud-images.ubuntu.com/releases/16.04/release/
+2. Keep a pristine copy of the image, in case you want to reset the machine, replacing <ubuntu-image> with the name of the file you downloaded on step 1:
+
+    ```
+    $ cp <ubuntu-image> <ubuntu-image>.pristine
+    ```
+
+3. Download the latest UEFI firmware image QEMU_EFI.fd from https://releases.linaro.org/components/kernel/uefi-linaro/latest/release/qemu64/
+4. Create a cloud init file, replacing <user> and <launchpad-user-name> with your values:
+
+    ```
+    $ cat > cloud-data.yaml << EOF
+    #cloud-config
+    users:
+      - name: <user>
+        ssh-import-id <launchpad-user-name>
+        sudo: ['ALL=(ALL) NOPASSWD:ALL']
+        groups: sudo
+        shell: /bin/bash
+    EOF
+    ```
+
+5. Create a cloud-config disk image:
+
+    ```
+    $ sudo apt install --yes cloud-image-utils
+    $ cloud-localds --disk-format qcow2 cloud.img cloud-data.yaml
+    ```
+
+6. Run the image in qemu, replacing <ubuntu-image> with the path of the file you downloaded on step 1.
+
+    ```
+    $ sudo apt install qemu-system-arm
+    $ qemu-system-aarch64 \
+        -smp 2 \
+        -m 1024 \
+        -M virt \
+        -cpu cortex-a57 \
+        -bios QEMU_EFI.fd \
+        -nographic \
+        -device virtio-blk-device,drive=image \
+        -drive if=none,id=image,file=<ubuntu-image> \
+        -device virtio-blk-device,drive=cloud \
+        -drive if=none,id=cloud,file=cloud.img \
+        -device virtio-net-device,netdev=user0 \
+        -netdev user,id=user0 \
+        -redir tcp:2222::22
+    ```
```

Comment:
Perhaps you can mention something like `There will be several error messages about missing modules, followed by complaints about the video mode and "Ubuntu 16.04" being rendered multiple times before the machine is booted.` here? It's rather obscure otherwise and made me wonder if it's broken and re-try from scratch. Apparently I had to wait a rather long time until it got to a login prompt and I could proceed to using SSH.